pkg/generic/generic_service.go

```diff
@@ -35,21 +36,39 @@ type Service interface {
 }
 
 // ServiceInfo create a generic ServiceInfo
-func ServiceInfo(pcType serviceinfo.PayloadCodec) *serviceinfo.ServiceInfo {
-	return newServiceInfo(pcType)
+func ServiceInfo(pcType serviceinfo.PayloadCodec, codec serviceinfo.CodecInfo) *serviceinfo.ServiceInfo {
+	return newServiceInfo(pcType, codec)
 }
 
-func newServiceInfo(pcType serviceinfo.PayloadCodec) *serviceinfo.ServiceInfo {
-	serviceName := serviceinfo.GenericService
+func newServiceInfo(pcType serviceinfo.PayloadCodec, codec serviceinfo.CodecInfo) *serviceinfo.ServiceInfo {
 	handlerType := (*Service)(nil)
-	methods := map[string]serviceinfo.MethodInfo{
-		serviceinfo.GenericMethod: serviceinfo.NewMethodInfo(callHandler, newGenericServiceCallArgs, newGenericServiceCallResult, false),
+
+	var methods map[string]serviceinfo.MethodInfo
+	var svcName string
+
+	if codec == nil {
+		svcName = serviceinfo.GenericService
```

Comment:
We need to have logic to get the real ServiceName for binary generic (= when codec == nil). Otherwise, a server cannot have multiple binary generic services.

Reply:
mark it as TODO

Reply:
Binary generic cannot be mixed with multi services. I think this can be considered as a default premise.